src/view/createPullRequestHelper.ts

```diff
@@ -37,28 +37,32 @@ export class CreatePullRequestHelper {
 			}
 		}));
 
-		this._disposables.push(this._createPRViewProvider!.onDidChangeSelectedBranch(selectedBranch => {
+		this._disposables.push(this._createPRViewProvider!.onDidChangeBaseBranch(selectedBranch => {
 			this._treeView?.updateBaseBranch(selectedBranch);
 		}));
 
-		this._disposables.push(this._createPRViewProvider!.onDidChangeSelectedRemote(remoteInfo => {
+		this._disposables.push(this._createPRViewProvider!.onDidChangeBaseRemote(remoteInfo => {
 			this._treeView?.updateBaseOwner(remoteInfo.owner);
 		}));
 	}
 
-	async create(extensionUri: vscode.Uri, folderRepoManager: FolderRepositoryManager, isDraft: boolean) {
+	async create(extensionUri: vscode.Uri, folderRepoManager: FolderRepositoryManager, compareBranch: string | undefined, isDraft: boolean) {
 		vscode.commands.executeCommand('setContext', 'github:createPullRequest', true);
 		if (!this._createPRViewProvider) {
 			const pullRequestDefaults = await folderRepoManager.getPullRequestDefaults();
 
-			this._createPRViewProvider = new CreatePullRequestViewProvider(extensionUri, folderRepoManager, pullRequestDefaults, !!isDraft);
-			this._treeView = new CompareChangesTreeProvider(this.repository, pullRequestDefaults.owner, pullRequestDefaults.base, folderRepoManager);
+			const branch = (compareBranch ? await folderRepoManager.repository.getBranch(compareBranch) : undefined) ?? folderRepoManager.repository.state.HEAD;
+			// TODO@eamodio what should we do if there is no valid branch?
```

Comment:
in what situations is `HEAD` undefined? I think it would be sane to error out here and show a notification, but we could look for any other local branches first

Reply:
In a rebase -- but not sure what the our Git api does in that case. It probably won't be undefined but probably won't be a valid branch either. Another case is a repo that has no branches yet, but that's an edge case